vars/deploy.groovy

```diff
@@ -41,9 +44,10 @@ def call(Map args = [:]) {
     spawn(image: image) {
       def userNS = Utils.usersNamespace();
       def deployNS = userNS + "-" + args.env;
-
+      pauseDeployments(deployNS, res.DeploymentConfig)
```

Comment:
that seems an unrelated change, pausing/resume,is it needed for oc binary move ?

Reply:
This is to avoid multiple deployment triggers. 

Problem:
Whenever pipeline tag an image from build to stage namespace it triggers new deployment. When we apply the DC with new image version it triggers another DC. All this happens as part of the deployment. We don't have control over how the user defines triggers in booster templates.
Hence multiple events triggering multiple deployments.

Solution:
Irrespective of how many changes we do in configuration let collectively apply the change in one go.
This pause and resume could fixes this issue. Before start making any change in the configuration we are pausing deployments, patching all configuration changes and resuming deployment.

Reply:
I think that's an issue with how the triggers works but I guess this is okay for now, is there any chance add this comment you made into the code so people understands it?

Reply:
yeah, its kind of mixing two changes in one PR. Will update it PR description, code comments and commit message.

I thought the change would be cheaper to do and related to `oc` in deploy context, so let it go in one PR.

Bur agrees with your comment.

Reply:
Cool, thanks, Let me know when its done so we can merge this!

Reply:
it may be better to have a method that accepts a closure and then  pauses, apply the steps (calls closure) and resumes e.g. 

```groovy

deployResources(ns, dc) {
  tagImage
  applyResources
}
```